src/compiler/binder.ts

```diff
@@ -843,9 +920,31 @@ namespace ts {
 
         function bindWorker(node: Node) {
             switch (node.kind) {
+                /* Strict mode checks */
                 case SyntaxKind.Identifier:
                     return checkStrictModeIdentifier(<Identifier>node);
                 case SyntaxKind.BinaryExpression:
+                    if (isAmdExportAssignment(node)) {

```

Comment:
Put all of this in an `if (isJavaScriptFile) { ... }` block. Right now we perform multiple tests that always produce false in a TypeScript file.